### ISSUE TITLE
Webpack 5 support

### ISSUE THREAD

**Issue by AdrienLemaire** [ph1p/i18next-scanner-webpack#11]
When upgrading webpack to v5, it breaks with an i18next-scanner-webpack error

```console
$ npx webpack-dev-server --config ./webpack.development.babel.js
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received an instance of Object
    at new NodeError (node:internal/errors:258:15)
    at validateString (node:internal/validators:123:11)
    at Object.isAbsolute (node:path:1029:5)
    at isModule (/home/dori/Projects/Work/GOunite/v3/client/node_modules/i18next-scanner-webpack/index.js:5:36)
    at /home/dori/Projects/Work/GOunite/v3/client/node_modules/i18next-scanner-webpack/index.js:71:18
    at Array.forEach (<anonymous>)
    at i18nextWebpackPlugin.apply (/home/dori/Projects/Work/GOunite/v3/client/node_modules/i18next-scanner-webpack/index.js:64:28)
    at createCompiler (/home/dori/Projects/Work/GOunite/v3/client/node_modules/webpack/lib/webpack.js:71:12)
    at create (/home/dori/Projects/Work/GOunite/v3/client/node_modules/webpack/lib/webpack.js:115:15)
    at webpack (/home/dori/Projects/Work/GOunite/v3/client/node_modules/webpack/lib/webpack.js:139:31)
```

Related config:
```js
const plugins = [
  new I18nextWebpackPlugin({
    dest: "./public/locales/",
    options: {
      func: {
        list: ["t"],
        extensions: [".js", ".jsx", ".ts", ".tsx"],
      },
      lngs: ["en", "ja"],
      resource: {
        loadPath: "./public/locales/{{lng}}/{{ns}}.json",
        savePath: "{{lng}}/{{ns}}.json",
      },
      // NOTE: This should contain all utilities/i18n.ts options + lazy loaded namespaces
      ns: [
        "admin",
        "auth",
        "common",
        "form",
        "glossary",
        "settings",
        "translation",
      ],
      defaultNs: "common",
    },
    transform: function customTransform(file, enc, done) {
      const { parser } = this;
      const content = fs.readFileSync(file.path, enc);
      let ns;
      const match = content.match(/useTranslation\(.+\)/);
      // QUESTION: Does someone know how to destructure the following line?
      if (match) ns = match[0].split(/('|")/)[2]; // eslint-disable-line prefer-destructuring
      let count = 0;
      parser.parseFuncFromString(content, { list: ["t"] }, (key, options) => {
        // https://github.com/i18next/i18next-scanner/issues/142#issuecomment-623848980
        let thisNs = ns;
        let myKey = key;
        if (/.+:.+/.test(key)) {
          [thisNs, myKey] = key.split(":");
        }
        parser.set(myKey, {
          ...options,
          ns: thisNs || "common",
          nsSeparator: ":",
          keySeparator: ".",
        });
        count += 1;
      });
      if (count > 0) {
        console.log(
          `i18next-scanner: count=${chalk.cyan(count)}, file=${chalk.yellow(
            JSON.stringify(file.relative)
          )}`
        );
      }

      done();
    },
  }),
};

export default {
  entry: {
    app: "./src/index.tsx",
  },
  plugins,
};
```

Debugging the code 
```js
class i18nextWebpackPlugin {
  apply(compiler) {
      const entry = compiler.options.entry;

      if (typeof entry === 'string') {
        this.i18nConfig.src = [path.dirname(entry)];
      } else if (typeof entry === 'object') {
        let entries = [];
        Object.keys(entry).forEach((e) => {
          const currentEntry = entry[e];
          let paths = [];

          if (Array.isArray(currentEntry)) {
            paths = currentEntry.filter((e) => !isModule(e)).map((e) => path.dirname(e));
          } else {
            console.dir(currentEntry);  // <== DEBUG PRINT currentEntry
            if (!isModule(currentEntry)) {  // <== FAILING LINE
              paths.push(path.dirname(currentEntry));
            }
          }

          entries = entries.concat(paths);
        });
```
```console
{ import: [ './src/index.tsx' ] }
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received an instance of Object
...
```

Tried to modify my entry as a string
```diff
-   entry: {
-     app: "./src/index.tsx",
-   },
+   entry: "./src/index.tsx",
```
But the result is the same, with currentEntry somehow equals to `{ import: [ './src/index.tsx' ] }`
printing `compiler.options.entry`, I get `{ main: { import: [ './src/index.tsx' ] } }`

**Comment by ph1p** [comment]
Thanks for reporting! I'll try to fix it as soon as possible.

**Comment by ph1p** [comment]
Should be fixed with the newer version "0.6.0".  In webpack 5 the entries have been renewed and it is now possible to specify different types. I hope that I have fixed it with the update for most cases. If you notice anything else, always feel free to post an issue (:
